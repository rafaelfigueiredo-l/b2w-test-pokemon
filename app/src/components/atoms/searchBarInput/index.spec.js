import SearchBarInput from ".";

describe("<SearchBarInput />", () => {
  const onChangeMock = jest.fn();

  const component = (
    <SearchBarInput onChange={onChangeMock()} value="ValueText">
      Teste
    </SearchBarInput>
  );

  const mounted = mount(component);
  const wrapper = shallow(component);

  it("should mount correctly SearchBarInput component", () => {
    expect(wrapper.find(".searchBarInput")).toHaveLength(1);
    expect(mounted).toMatchSnapshot();
  });

  it("should unmount correctly SearchBarInput component", () => {
    expect(mounted.unmount()).toEqual({});
  });

  it("should render input type checkbox", () => {
    expect(wrapper.find("[type='text']")).toHaveLength(1);
  });

  it("should call onChange function", () => {
    expect(onChangeMock).toHaveBeenCalledTimes(1);
    expect(onChangeMock).toBeCalled();
  });
});
