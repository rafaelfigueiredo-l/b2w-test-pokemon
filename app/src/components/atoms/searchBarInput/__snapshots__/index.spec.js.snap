// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBarInput /> should mount correctly SearchBarInput component 1`] = `
<styled.input
  className="searchBarInput"
  placeholder="Procurar Pokemon..."
  type="text"
>
  <input
    className="sc-AxjAm fBBAip searchBarInput"
    placeholder="Procurar Pokemon..."
    type="text"
  />
</styled.input>
`;
